javaslang/src/main/java/javaslang/collection/Vector.java

```diff
@@ -694,10 +694,10 @@ public int indexOf(T element, int from) {
 
     @Override
     public Vector<T> init() {
-        if (isEmpty()) {
-            throw new UnsupportedOperationException("init of empty Vector");
-        } else {
+        if (nonEmpty()) {
```

Comment:
nonEmpty() has one more method call then isEmpty(), but I think you say because of JIT magic it does not matter because it is inlined.

What is the latency of inlining? I think it will cost something! The effort can't be zero. I see trusting in inlining like blurring the reality.